src/amplify-migration/amplifyMigrationScript.ts

```diff
@@ -0,0 +1,289 @@
+  await simpleGit(repoPath).commit(commitMessage);
+}
+
+/**
+ * Requirements:
+ * convert `permalink: /some/path/` to `some/path`
+ * convert `/some/path` to `some/path`
+ * convert `some/path/` to `some/path`
+ * @param permalink original permalink
+ * @returns raw permalinks without leading/trailing slash
+ */
+function getRawPermalink(permalink: string) {
+  let trimmedPermalink = permalink.trim();
+  if (trimmedPermalink.startsWith(`permalink: `)) {
+    trimmedPermalink = permalink.trim().slice(11);
+  }
+  if (trimmedPermalink.startsWith(`/`)) {
+    trimmedPermalink = trimmedPermalink.slice(1);
+  }
+  if (trimmedPermalink.endsWith("/")) {
+    trimmedPermalink = trimmedPermalink.slice(0, -1);
+  }
+
+  return trimmedPermalink;
+}
+
+async function changePermalinksInFiles(
+  filePath: string,
+  repoPath: string,
+  changedPermalinks: { [key: string]: string }
+) {
+  // read file content
+  let fileContent = await fs.promises.readFile(filePath, "utf-8");
+  let fileChanged = false;
+
+  ({ fileContent, fileChanged } = changeFileContent(
+    fileContent,
+    changedPermalinks
+  ));
+
+  if (fileChanged) {
+    await fs.promises.writeFile(filePath, fileContent, "utf-8");
+    await simpleGit(repoPath).add(filePath);
+  }
+}
+
+function changeFileContent(
+  fileContent: string,
+  changedPermalinks: { [key: string]: string }
+) {
+  let fileChanged = false;
+  // three different regex patterns to take care of
+  // 1. href="original_permalink"
+  // 2. [click here](original_permalink)
+
+  const markdownRegex = /\[(.*?)\]\((.*?)\)/g;
+  const dom = new JSDOM(fileContent);
+  dom.window.document.querySelectorAll("a").forEach((a: any) => {
+    //replace permalinks with lowercase and in changedPermalinks
+    let rawPermalink = getRawPermalink(a.href);
+
+    if (changedPermalinks[rawPermalink]) {
+      a.href = a.href.replace(rawPermalink, changedPermalinks[rawPermalink]);
+      fileChanged = true;
+    }
+  });
+
+  fileContent = fileChanged ? dom.window.document.body.innerHTML : fileContent;
+  const markdownMatches = fileContent.match(markdownRegex);
+
+  if (markdownMatches) {
+    for (const match of markdownMatches) {
+      console.log(match);
+      let originalPermalink = match.slice(match.indexOf("(") + 1, -1);
+      originalPermalink = getRawPermalink(originalPermalink);
+      if (!changedPermalinks[originalPermalink]) {
+        continue;
+      }
+      const newPermalink = originalPermalink.toLocaleLowerCase();
+      const newMatch = match.replace(originalPermalink, newPermalink);
+      fileContent = fileContent.replace(match, newMatch);
+
+      fileChanged = true;
+    }
+  }
+  console.log(fileContent);
+  return { fileContent, fileChanged };
+}
+
+async function updateConfigYml(appId: string, repoPath: string) {
+  const configFile = path.join(repoPath, `_config.yml`);
+
+  // Read the file into a variable
+  let configYml = await fs.promises.readFile(configFile, { encoding: "utf-8" });
```

Comment:
suggest changing to `configYmlContents` for clarity

Reply:
7c9bfd9